### ISSUE TITLE
[BUG] Qt6 books list scrolls but Qt5 stacks

### ISSUE THREAD

**Issue by B-Interactive** [bibletime/bibletime#476]
**Bug description**
If I build for Qt6, the bible "Select book" drop-down may be larger than the screen, depending on resolution, such as 1440p or lower.  Not all books are visible and on first click of this menu, scrolling is often not available so books like Genesis and Revelation will not be available.  Re-clicking the menu seems to make scrolling available, but it remains a more difficult means of accessing books.

![Screenshot_20240909_120937](https://github.com/user-attachments/assets/760bb827-97d2-4290-b8f3-33f05708cb23)

**Steps to reproduce**
1. Built with the following build dependencies (explicitly present during build):
git, po4a, docbook-xsl, docbook-xml, clucene, cmake, qt6-tools
2. Built with the following dependencies (explicitly persist after build):
sword, qt6-webengine, qt6-svg
3. Built with the following cmake commands:
```
-DCMAKE_INSTALL_PREFIX=/usr \
-DCMAKE_BUILD_TYPE=Release \
-DUSE_QT6=ON \
-DBUILD_HANDBOOK_HTML=ON \
-DBUILD_HANDBOOK_PDF=OFF \
-DBUILD_HOWTO_HTML=ON \
-DBUILD_HOWTO_PDF=OFF \
-DBT_DOCBOOK_XSL_HTML_CHUNK_XSL=$_xslstylespath/html/chunk.xsl \
-DBT_DOCBOOK_XSL_PDF_DOCBOOK_XSL=$_xslstylespath/fo/docbook.xsl \
-DQT_LUPDATE_EXECUTABLE=/usr/lib/qt6/bin/lupdate \
-DQT_LRELEASE_EXECUTABLE=/usr/lib/qt6/bin/lrelease \
```
4. After build, launch, install a translation and test by clicking the "Select book" dropdown.

**Expected behavior**
If I build for Qt5, the bible "Select book" drop-down will not produce a menu that is larger than the screen.  It will stack the menu to the right, so all books are visible and accessible.

![Screenshot_20240909_120116](https://github.com/user-attachments/assets/e5441a77-187f-449c-926e-00c62e6878da)

**System information:**
 - EndeavourOS (Arch based) up-to-date rolling release.  Reproduced in a clean install VM.
 - BibleTime version b3ff1d542754a87aae44a4bf1f2bb4739b13f191
 - Installed from source, [using PKGBUILD scripts](https://aur.archlinux.org/packages/bibletime-git) modified for testing against Qt5 and Qt6
 - Sword 1.9.0
 - Qt 6.7.2

**Additional context**
The Qt5 test is built as such:
1. Using the following build dependencies:
sword, qt5-webengine, qt5-svg
2. Using the following dependencies:
git, po4a, docbook-xsl, docbook-xml, clucene, cmake, qt5-tools
3. Using the following cmake commands:
```
-DCMAKE_INSTALL_PREFIX=/usr \
-DCMAKE_BUILD_TYPE=Release \
-DUSE_QT6=OFF \
-DBUILD_HANDBOOK_HTML=ON \
-DBUILD_HANDBOOK_PDF=OFF \
-DBUILD_HOWTO_HTML=ON \
-DBUILD_HOWTO_PDF=OFF \
-DBT_DOCBOOK_XSL_HTML_CHUNK_XSL=$_xslstylespath/html/chunk.xsl \
-DBT_DOCBOOK_XSL_PDF_DOCBOOK_XSL=$_xslstylespath/fo/docbook.xsl \
```

**Comment by michaelweghorn** [comment]
I can reproduce with the Qt 6 build on Debian testing in a KDE Plasma dev session, but for me, this only happens on Wayland.

If I start with environment variable `QT_QPA_PLATFORM=xcb` set to force running as X client on XWayland instead, the first entry aligns properly with the top of the screen.

I can reproduce similar behavior with this simple Qt program using a `QToolButton`  (which is the base class for `BtDropdownChooserButton` as well):

```
#include <QApplication>
#include <QMainWindow>
#include <QMenu>
#include <QToolButton>

int main(int argc, char *argv[])
{
    QApplication a(argc, argv);

    QToolButton toolButton;
    QMenu* menu = new QMenu;
    for (int i = 0; i < 100; i++)
    {
        menu->addAction("Entry " + QString::number(i+1));
    }
    toolButton.setMenu(menu);
    toolButton.setPopupMode(QToolButton::InstantPopup);
    toolButton.show();


    return a.exec();
}
```

This makes me think that at least the issue I'm seeing may be more likely a bug in Qt or the Qt style in use (which is Breeze by default for me, but it's similarly broken with `QT_STYLE_OVERRIDE=Fusion`).

However, another issue I'm seeing with BibleTime only, not the above sample program, is that scrolling through the list of books doesn't work when clicking the button for the first time, but only after closing the popup and re-opening it by clicking the button again. (That's the same with `QT_QPA_PLATFORM=xcb` however).

**Comment by michaelweghorn** [comment]
Upstream Qt bug report: https://bugreports.qt.io/browse/QTBUG-137955
